reporter/internal/pdata/generate.go

```diff
@@ -29,16 +30,27 @@ const (
 // Generate generates a pdata request out of internal profiles data, to be
 // exported.
 func (p *Pdata) Generate(tree samples.TraceEventsTree,
-	agentName, agentVersion string) pprofile.Profiles {
+	agentName, agentVersion string) (pprofile.Profiles, error) {
 	profiles := pprofile.NewProfiles()
+	dic := profiles.ProfilesDictionary()
+
+	// Temporary helpers that will build the various tables in ProfilesDictionary.
+	stringSet := libpf.OrderedSet[string]{}
+	funcSet := libpf.OrderedSet[samples.FuncInfo]{}
+	mappingSet := libpf.OrderedSet[libpf.FileID]{}
+
+	// By specification, the first element should be empty.
+	stringSet.Add("")
+	funcSet.Add(samples.FuncInfo{Name: "", FileName: ""})
 
 	for containerID, originToEvents := range tree {
 		if len(originToEvents) == 0 {
 			continue
 		}
 
 		rp := profiles.ResourceProfiles().AppendEmpty()
-		rp.Resource().Attributes().PutStr(string(semconv.ContainerIDKey), string(containerID))
+		rp.Resource().Attributes().PutStr(string(semconv.ContainerIDKey),
```

Comment:
Unrelated to this PR (but a low-hanging fruit): use `ResourceProfiles().EnsureCapacity()` before the loop.